Topics/Tech_Stacks/Writing_Clean_React_Code.md

```diff
@@ -0,0 +1,153 @@
+
+- **Variables and Functions**: Variables and functions should be named using camelCase. For example, a variable that stores a user's name should be named `userName` instead of `user_name` or `user-name`, and a function that handles a button click should be named `handleClick` instead of `handle_click` or `handle-click`. This convention is consistent with the naming of built-in React variables such as `onClick` and `onChange` and built-in React functions such as `useState` and `useEffect`.
+
+- **State Variables and Setters**: State variables should be prefixed with _is_, _has_, or _should_ to denote boolean values and the setters should be prefixed with _set_ followed by the name of the variable. For example, a state variable that stores whether a user is logged in should be named `isLoggedIn` and its setter should be named `setIsLoggedIn`.
+
+- **Event Handlers**: Event handlers should be named using camelCase and prefixed with _handle_. For example, a function that handles a button click should be named `handleClick` instead of `onClick`.`
+
+- **Props**: Props should be named using camelCase, similar to variables. Moreover, it is good practice to list props out when defining a component. This helps developers quickly identify the props that a component accepts. For example, a component that renders a button should be defined as follows:
+
+  ```jsx
+  const Button = ({ text, onClick }) => {
+    return <button onClick={onClick}>{text}</button>;
+  };
+  ```
+
+Notice that the onClick handler isn't defined in the component, and prop name isn't prefixed by _handle_. This is because the component is only responsible for rendering the button and not handling the button click. The onClick handler should be defined in the parent component that uses the Button component, and _in the parent component_ the event handler that is passed down to the Button component should be prefixed by _handle_. Like in the below example:
+
+```jsx
+const ParentComponent = () => {
+  const handleClick = () => {
+    // handle button click
+  };
+  return <Button text="Click Me" onClick={handleClick} />;
+};
+```
+
+- **Constants**: Constants should be named using uppercase letters and underscores to separate words. For example, a constant that stores user emails should be named `USER_EMAILS` instead of `user-emails` or `userEmails`.
+
+- **File Names**: File names should be named using PascalCase. For example, a file that contains a component that renders a navbar element should be named `NavbarElement.jsx` instead of `navbarElement.jsx` or `navbar-element.jsx`.
+
+- **CSS Classes and Files**: CSS classes should be named using kebab-case or camelCase. For example, a class that styles a div that acts as a container for a cards should be named `cards-container` or `cardsContainer` instead of `cards_container`. CSS Files follows the same convention as CSS classes.
```

Comment:
I believe what you're referring to here are HTML classes which are selected in CSS files, not named, so I think it'd be less confusing to talk about HTML classes, not "CSS classes"

Reply:
I believe classes can be referred to in both ways. Here is an excerpt from blog.hubspot: "What is a CSS class? A CSS class is an attribute used to define a group of HTML elements in order to apply unique styling and formatting to those elements with CSS."